contrib/syslinux-update.sh

```diff
@@ -0,0 +1,116 @@
+## initramfs, make sure to set the `Components.ImageDir` option in your
+## ZFSBootMenu configuration to a path within the syslinux file system.
+##
+## Ensure the file is executable and place it in the directory specified by the
+## `Global.PostHooksDir` option in your ZFSBootMenu configuration file.
+##
+## This script uses the SYSLINUX_ROOT to refer to the mount point of the
+## syslinux filesystem and KERNEL_PATH to be the subdirectory, relative to
+## SYSLINUX_ROOT, where ZFSBootMenu will install initramfs and kernel pairs.
+## Kernels are expected to have file names of the form
+##
+##     ${KERNEL_PREFIX}[-<version>]
+##
+## and a corresponding initramfs image
+##
+##     initramfs[-<version>].img
+##
+## When creating syslinux entries, the variable ZBM_KCL_ARGS defines the
+## command-line arguments that will be passed to each ZFSBootMenu kernel.
+##
+## Change these variables as necessary to reflect your configuration.
+
+## By default, syslinux is on /boot/syslinux
+SYSLINUX_ROOT="/boot/syslinux"
+## By default, kernel and initramfs pairs will be in /boot/syslinux/zbm
+KERNEL_PATH="zbm"
+## By default, kernels are files starting with "vmlinuz"
+KERNEL_PREFIX="vmlinuz"
+
+# By default, no arguments are passed to the kernel
+ZBM_KCL_ARGS=""
+
+if [ ! -d "${SYSLINUX_ROOT}" ]; then
+	echo "ERROR: syslinux root '${SYSLINUX_ROOT}' does not exist"
+	exit 1
+fi
+
+KERNEL_DIR="${SYSLINUX_ROOT}/${KERNEL_PATH}"
+KERNEL_DIR="${KERNEL_DIR%/}"
+if [ ! -d "${KERNEL_DIR}" ]; then
+	echo "ERROR: kernel path '${KERNEL_DIR}' does not exist"
+	exit 1
+fi
+
+SYSLINUX_CFG=
+cleanup() {
+	[ -n "${SYSLINUX_CFG}" ] && rm -f "${SYSLINUX_CFG}"
+	unset SYSLINUX_CFG
+}
+
+trap cleanup EXIT INT TERM QUIT
+
+if ! SYSLINUX_CFG="$(mktemp)"; then
+	echo "ERROR: failed to make a temporary syslinux.cfg"
+	exit 1
+fi
+
+# Write the standard configuration header
+cat << EOF
+UI menu.c32
+PROMPT 0
+
+MENU TITLE Choose a ZFSBootMenu image to boot
+TIMEOUT 50
+EOF
+
+# Sort list of candidate kernels by version, newest first
+readarray -t KERNELS < <(printf '%s\n' "${KERNEL_DIR}/${KERNEL_PREFIX}"* | sort -V -r)
+
+# Identify each kernel/initramfs pair
+# The first in sort order will be the default
+DEFAULT_SET=
+for kern in "${KERNELS[@]}"; do
+	# Make sure file exists and strip leading path
+	[ -f "${kern}" ] || continue
+	kern="${kern##*/}"
+
+	# Strip the kernel prefix to look for matching initramfs
+	version="${kern#"${KERNEL_PREFIX}"}"
+	initramfs="initramfs${version}.img"
+	[ -f "${KERNEL_DIR}/${initramfs}" ] || continue
+
+	zbmlabel="zfsbootmenu${version//[[:space:]]/}"
+
+	if [ -z "${DEFAULT_SET}" ]; then
+		echo "DEFAULT ${zbmlabel}" >> "${SYSLINUX_CFG}"
+		DEFAULT_SET="yes"
+	fi
+
+	cat >> "${SYSLINUX_CFG}" <<-EOF
+
+	LABEL ${zbmlabel}
+	MENU LABEL ZFSBootMenu (${version#-})
+	LINUX ${KERNEL_PATH:+/${KERNEL_PATH}}/${kern}
+	INITRD ${KERNEL_PATH:+/${KERNEL_PATH}}/${initramfs}
+	APPEND ${ZBM_KCL_ARGUMENTS}
```

Comment:
This doesn't look like it's defined anywhere / is superseded by `ZBM_KCL_ARGS`

Reply:
Thanks, fixed